frontend/workflows/k8s/src/update-probe.tsx

```diff
@@ -0,0 +1,385 @@
+import React from "react";
+import type { clutch as IClutch } from "@clutch-sh/api";
+import {
+  Button,
+  ButtonGroup,
+  client,
+  Confirmation,
+  MetadataTable,
+  Resolver,
+  Select,
+  useWizardContext,
+} from "@clutch-sh/core";
+import { useDataLayout } from "@clutch-sh/data-layout";
+import type { WizardChild } from "@clutch-sh/wizard";
+import { Wizard, WizardStep } from "@clutch-sh/wizard";
+import { string } from "yup";
+
+import type { ConfirmChild, ResolverChild, WorkflowProps } from ".";
+
+// Examples of valid Seconds: 0.1, 100s
+const SECONDS_REGEX = /^([-+]?[0-9]*[sS]*)$/;
+
+const DeploymentIdentifier: React.FC<ResolverChild> = ({ resolverType }) => {
+  const { onSubmit } = useWizardContext();
+  const deploymentData = useDataLayout("deploymentData");
+  const inputData = useDataLayout("inputData");
+
+  const onResolve = ({ results, input }) => {
+    deploymentData.assign(results[0]);
+    inputData.assign(input);
+    onSubmit();
+  };
+
+  return <Resolver type={resolverType} searchLimit={1} onResolve={onResolve} />;
+};
+
+function findContainer(args: {
+  deploymentSpec: IClutch.k8s.v1.Deployment.IDeploymentSpec;
+  containerName: string;
+}): IClutch.k8s.v1.Deployment.DeploymentSpec.PodTemplateSpec.PodSpec.IContainer {
+  return args.deploymentSpec.template.spec.containers.find(
+    container => container.name === args.containerName
+  );
+}
+
+const DeploymentDetails: React.FC<WizardChild> = () => {
+  const { onSubmit, onBack } = useWizardContext();
+  const deploymentData = useDataLayout("deploymentData");
+  const deployment = deploymentData.displayValue() as IClutch.k8s.v1.Deployment;
+  const update = (key: string, value: boolean) => {
+    deploymentData.updateData(key, value);
+  };
+
+  const currentDeploymentData = useDataLayout("currentDeploymentData");
+
+  const { containers } = deployment.deploymentSpec.template.spec;
+
+  const [containerName, setContainerName] = React.useState(containers[0].name);
```

Comment:
Might want to put a check in here for the default in case containers is empty

Reply:
Adding validation of container array